### PR TITLE
do not use private AccountManager in UsersController

### PR DESCRIPTION
**Pull request by blizzz**
- extends IAccountProperty for verificationData getters and setters
- implementation thereof ^
- and of course adaption of UsersController

contributes to #26866

### CODE REVIEW

**Review by nickvergessen — APPROVED** [review]
LGTM

File: apps/settings/lib/Controller/UsersController.php [hunk 1]
```diff
@@ -393,86 +395,62 @@ public function setUserSettings(?string $avatarScope = null,
...
+			IAccountManager::PROPERTY_EMAIL => ['value' => $email, 'scope' => $emailScope],
+			IAccountManager::PROPERTY_WEBSITE => ['value' => $website, 'scope' => $websiteScope],
+			IAccountManager::PROPERTY_ADDRESS => ['value' => $address, 'scope' => $addressScope],
+			IAccountManager::PROPERTY_PHONE => ['value' => $phone, 'scope' => $phoneScope],
+			IAccountManager::PROPERTY_TWITTER => ['value' => $twitter, 'scope' => $twitterScope],
+		];
+		$allowUserToChangeDisplayName = $this->config->getSystemValueBool('allow_user_to_change_display_name', true);
+		foreach ($updatable as $property => $data) {
+			if ($allowUserToChangeDisplayName === false
+				&& in_array($property, [IAccountManager::PROPERTY_DISPLAYNAME, IAccountManager::PROPERTY_EMAIL])) {
```
Comment (PVince81) [inline apps/settings/lib/Controller/UsersController.php:413]: as far as  remember we always want to use `in_array` in strict mode, please add `true` as a third argument

**Review by PVince81 — APPROVED** [review]
:+1: looks fine

please consider my minor comment about in_array

**Review by juliusknorr — APPROVED** [review]

File: apps/settings/lib/Controller/UsersController.php (hunk 1 above)
Comment (blizzz) [reply apps/settings/lib/Controller/UsersController.php]: good catch, adjusted

**Pull request merged**